models/dvmdostem/R/write.config.dvmdostem.R

```diff
@@ -0,0 +1,353 @@
+#-------------------------------------------------------------------------------
+# Copyright (c) 2012 University of Illinois, NCSA.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the
+# University of Illinois/NCSA Open Source License
+# which accompanies this distribution, and is available at
+# http://opensource.ncsa.illinois.edu/license.html
+#-------------------------------------------------------------------------------
+
+
+##------------------------------------------------------------------------------------------------#
+##' convert parameters, do unit conversions and update parameter names from PEcAn database default
+##' to units/names within dvmdostem
+##'
+##' Performs model specific unit conversions on a a list of trait values,
+##' such as those provided to write.config
+##'
+##' @name convert.samples.dvmdostem
+##' @title Convert samples for dvmdostem
+##' @param trait.samples a matrix or dataframe of samples from the trait distribution
+##' @return matrix or dataframe with values transformed
+##' @export
+##' @author Shawn Serbin, Tobey Carman
+##' @importFrom udunits2 ud.convert
+
+convert.samples.dvmdostem <- function(trait_values) {
+
+  # convert to a data frame for easy access by column name.
+  if(is.list(trait_values)){
+    trait_values <- as.data.frame(trait_values)
+  }
+
+  # Rename variables - might be able to make some changes here
+  # that make the loops in write.config.dvmdostem easier to 
+  # understand/deal with
+  trait_names <- colnames(trait_values)
+  # for debugging
+  PEcAn.logger::logger.info(trait_names)
+  
+  # Not really sure how this should work yet? 
+  # Example from FATES model
+  # trait.names[trait.names == "leaf_respiration_rate_m2"]   <- "atref.rd"
+  #                                pecan notion?                dvmdostem?
```

Comment:
Correct, this would be an example of changing a name from PEcAn standard to model-specific